### DISCUSSION TITLE
Wyze cam v3 doesn't detect local network

### DISCUSSION THREAD

**Discussion by MatthewTingum** [gtxaspec/wz_mini_hacks#589] [General]
I bring this discussion here for a few reasons:
- I have modified the firmware to include a new root user with known credentials
  - This may have caused my issue
- I have a UART connection active
  - That again, may be causing this issue
- We may be able to diagnose this issue further with a root busybox shell
  - I have a more fully featured busybox than the one running on the camera

## Background

I have a Wyze Cam V3 (Not Pro) running firmware version 4.36.10.4054.

I wanted code execution so I started by chip clipping the SPI flash. No luck detecting the flash, so I lifted it and read the SPI out of circuit. I then added a user with creds `user:password`. After replacing the SPI flash I was successful logging in over UART with the creds `user:password`. At the time, I was able to upload my own copy of busybox, launch a telnet daemon, and go wild.

I put together a similar setup for a friend so we could poke around the Wyze together. The only problem was that the 2nd camera I modified would not connect to my wifi. I was left thinking that I might have damaged something in the process.

A day passes and I'm no longer able to connect to wifi on the first camera I had modified. Weird... I hadn't really moved the camera and the solder joints to the UART and SPI flash are pretty steady. This pushed me to finally replace the router I'm renting from an ISP with my own router. After doing so it worked again! I just had to hook up UART again and launch the telnet daemon. I hooked up UART on the live system but got no response. No worries, I'll just restart the camera. After restarting the camera, I was able to interact with the UART shell. But this time, it refused to connect to the wifi again.

## How can this be fixed?

Maybe I try to change my SSID again. I don't know exactly what fixed it between router swaps. One thing I do know is that the SSID changed. It has successfully connected to the first and second router at one point.

???

I'm open to suggestions. Curious to know if others have run into this.

**Comment by gtxaspec** [comment]
Did you make a backup of the original flash?

Also, you could have used wz_mini to root the FW without modifying the flash... even on newer firmware.  Just FYI =D

**Reply by MatthewTingum** [reply]
Yes, I have a flash dump as it came from the factory (never powered on), and a dump from after the firmware was updated and connected to wifi.

That... would have been good to know lol. I had a vague understanding that code execution could be gained on older firmwares but never really considered that one could downgrade. Even so, I have a socket for the SPI flash and am enjoying my setup :man_shrugging:

I'm going to attempt an antenna mod. I'm not sure that it will give better results than setting the camera a foot away from the router but who knows.

**Reply by MatthewTingum** [reply]
Do you know how to download firmware directly? I'd imagine the cam asks for the latest version, compares against it's own, then prompts to download if a newer version is available.

**Reply by gtxaspec** [reply]
if you check uboot as the camera boots up, it looks for a kernel, `factory_t31_ZMC6tiIDQN` or an update file `demo_wcv3.bin`

the demo file is checked against a signature in u-boot, so its easier to setup a kernel with a custom initramfs or something like that to flash... or just load wz_mini and boot, and then flash from there.

**Reply by gtxaspec** [reply]
and since you have the camera setup for a dev environment, check out OpenIPC, a full replacement linux system with support for the wyze v3.  www.openipc.org

I just setup a few camera's and they are working okay so far.

**Reply by endertable** [reply]
@gtxaspec so is openipc something you would recommend just for the cameras that can't connect to the Wyze infrastructure anymore or is this something that is better than what Wyze currently has, e.g. more or better streaming options, etc.

**Reply by gtxaspec** [reply]
If you want a cloudless, local oriented device, then openipc would be the choice.  All it does is offer video streams so you can either view or process with something else, aka frigate or home assistant.

I'm trying to make an easy to install version soon.  Currently setup and install is geared towards developers as it requires opening the camera.

**Reply by MatthewTingum** [reply]
My camera decided to have reliable connection to my router again. IDK what changed.

Have you considered using `flashrom` for your easy install? I'm imagining a scenario where a user gains initial code execution via SD card. Then they can run flashrom to flash a permanent image.

You could also provide release packages that users can copy to an SD card. Assume the user keeps the SD card inserted at all times.

I apologize if my suggestions have glaring issues. I *still* haven't read through everything.
